WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCaseTest.kt

```diff
@@ -0,0 +1,128 @@
+package org.wordpress.android.ui.jetpack.backup.download.usecases
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.flow.toList
+import org.assertj.core.api.Assertions
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.BackupDownloadStatusModel
+import org.wordpress.android.fluxc.store.ActivityLogStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.OnBackupDownloadStatusFetched
+import org.wordpress.android.test
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadRequestState.Failure
+import org.wordpress.android.util.NetworkUtilsWrapper
+import org.wordpress.android.fluxc.action.ActivityLogAction.FETCH_BACKUP_DOWNLOAD_STATE
+import org.wordpress.android.fluxc.store.ActivityLogStore.BackupDownloadStatusError
+import org.wordpress.android.fluxc.store.ActivityLogStore.BackupDownloadStatusErrorType.GENERIC_ERROR
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadRequestState.Complete
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadRequestState.Empty
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadRequestState.Failure.RemoteRequestFailure
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadRequestState.Progress
+import java.util.Date
+
+@InternalCoroutinesApi
+class GetBackupDownloadStatusUseCaseTest : BaseUnitTest() {
+    private lateinit var useCase: GetBackupDownloadStatusUseCase
+    @Mock lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock lateinit var activityLogStore: ActivityLogStore
+    @Mock private lateinit var site: SiteModel
+    @Mock lateinit var statusModel: BackupDownloadStatusModel
```

Comment:
Suggestion (💡 ): Consider stubbing the models instead of mocking them. Usually it is better to use fakes for data related classes and only use mocks for logic related classes.